Software/app/basic_bootloader/README.md

```diff
@@ -0,0 +1,89 @@
+# basic_bootloader
+
+## Description
+
+The application provides a basic bootloader that can be combined with other applications to perform a specific functionality at boot time.
+
+> If you want to learn more about the purpose of a bootloader, we recommend reading this great [blog post](chttps://interrupt.memfault.com/blog/how-to-write-a-bootloader-from-scratch).
+
+`basic_bootloader` performs one of two functionalities based on the GNSE button:
+
+1. If the button is not pressed -> boots the user flashed application
+2. If the button is pressed long enough -> boots the ST internal bootloader
+
+## Configuration
+
+`basic_bootloader` key configurations:
+
+### Memory Map
+
+The internal flash (ROM) memory map can be configured to accommodate a bootloader and an application by adjusting the parameters defined in [`memory_map.ld`](./../../target/memory_map.ld).
+
+```
+BOOTROM  (rx)  : ORIGIN = 0x08000000, LENGTH = 0x00000B000   /* Flash memory dedicated to bootloader */
+APPROM   (rx)  : ORIGIN = 0x0800B000, LENGTH = 0x000035000   /* Flash memory dedicated to application */
+```
+
+### App activity
+
+The application behavior can be adjusted by modifying [`conf/app_conf.h`](./conf/app_conf.h).
+
+- `GNSE_TINY_TRACER_ENABLE` enables UART (115200/8-N-1) logging of application activity
+
+```c
+#define GNSE_TINY_TRACER_ENABLE 1
+```
+
+> Please note that it is considered as a best practice to disable logging when building bootloaders to reduce the binary size.
+
+## Setup
+
+To build a `baic_bootloader` that can be combined with another application in the same memory, you need to use a linker script that supports the new memory map.
+
+### Building with CMake
+
+If you are building using CMake, then update [`CMakeLists.txt`](./../../CMakeLists.txt) `LINKER_SCRIPT` flag to `bootloader.ld` when compiling a `basic_bootloader` binary.
```

Comment:
Maybe link for bootloader.ld could be added

Reply:
This is intentional, see this line
https://github.com/TheThingsIndustries/generic-node-se/blob/92cf8069113548399029a2c4555a5e38ed44e5d2/Software/CMakeLists.txt#L58